### PR TITLE
Disputable app PoC

### PR DESCRIPTION
**Pull request by facuspagnuolo**

### CODE REVIEW

File: contracts/Delay.sol [hunk 1]
```diff
@@ -64,53 +69,38 @@ contract Delay is AragonApp, IForwarder {
...
+    * @param _context Information context for the script being scheduled
     * @param _evmCallScript The script that can be executed after a delay
     */
-    function delayExecution(bytes _evmCallScript) external auth(DELAY_EXECUTION_ROLE) returns (uint256) {
-        return _delayExecution(_evmCallScript);
-    }
-
-    function isForwarder() external pure returns (bool) {
-        return true;
+    function delayExecution(bytes _context, bytes _evmCallScript) external auth(DELAY_EXECUTION_ROLE) returns (uint256) {
```
Comment (sohkai) [inline contracts/Delay.sol:75]: Note that we will want to be careful to be backwards compatible here.

File: contracts/Delay.sol [hunk 2]
```diff
@@ -148,22 +139,81 @@ contract Delay is AragonApp, IForwarder {
...
+    function _onDisputableAllowed(uint256 _delayedScriptId) internal {
+        _resumeExecution(_delayedScriptId);
+    }
+
+    /**
+    * @dev Reject script execution
+    * @param _delayedScriptId The ID of the script execution to reject
+    */
+    function _onDisputableRejected(uint256 _delayedScriptId) internal {
+        require(_isExecutionPaused(_delayedScriptId), ERROR_SCRIPT_NOT_PAUSED);
```
Comment (sohkai) [inline contracts/Delay.sol:166]: Here and implicitly in the other hooks, we have some requires.

I wonder if we shouldn't treat these hooks as "always safe", as I assume reverting there would be really bad (potentially deadlocking that court case?)

Comment (bingen) [reply contracts/Delay.sol:166]: Interesting question, although not sure what’s best here. Not reverting may be bad too, leaving contracts in inconsistent states.

**Comment by facuspagnuolo** [comment]
Closing in favor of #60

**Pull request closed without merging**